### PR TITLE
[8.x] PHP 8.1 Support

### PR DESCRIPTION
**Pull request by driesvints**
This PR contains more PHP 8.1 fixes:

- Remove mimicking PHP 8.0 now that we're nearing the PHP 8.1 release date
- Add `prefer-lowest` builds
- Add separate build step for minimum versions for Symfony, PHPUnit & CommonMark we cannot bump these through Composer
- Bumps some libraries for PHP 8.1 compatibility
- Added more clearer message for skipped mail tests on PHP 8.1 since SwiftMailer won't be patched for it
- Unskipped PHP 8.1 tests

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -68,6 +61,14 @@ jobs:
           command: composer require guzzlehttp/guzzle:^7.2 --no-interaction --no-update
         if: matrix.php >= 8
 
+      - name: Set Minimum Symfony Version
+        uses: nick-invision/retry@v1
+        with:
+          timeout_minutes: 5
+          max_attempts: 5
+          command: composer require symfony/symfony:5.4.x-dev --no-interaction --no-update
```
Comment (GrahamCampbell) [inline .github/workflows/tests.yml:69]: why do we need this?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (driesvints) [reply .github/workflows/tests.yml]: PHP 8.1 will only be fully support on Symfony 5.4. This is needed for prefer-lowest builds without bumping the minimum Symfony requirement for non-PHP 8.1 installs.

Comment (GrahamCampbell) [reply .github/workflows/tests.yml]: > PHP 8.1 will only be fully support on Symfony 5.4.

Oh, really? I thought Symfony treated PHP version support as a bug fix?

Comment (driesvints) [reply .github/workflows/tests.yml]: Nicolas confirmed this with me.

**Comment by tillkruss** [comment]
Predis fix merged here: https://github.com/predis/predis/commit/92adf32e88cc7e86e30dd073ecefefff0547fdb3

File: composer.json [hunk 2]
```diff
@@ -89,7 +89,7 @@
         "orchestra/testbench-core": "^6.23",
         "pda/pheanstalk": "^4.0",
         "phpunit/phpunit": "^8.5.19|^9.5.8",
-        "predis/predis": "^1.1.8",
+        "predis/predis": "v1.1.x-dev",
```
Comment (GrahamCampbell) [inline composer.json:92]: `^1.1.9`

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (driesvints) [reply composer.json:92]: I'll do that when Till has actually tagged the release.

Comment (GrahamCampbell) [reply composer.json:92]: why? it'll work even without, and it means we don't need to go back and update the code later.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by driesvints** [comment]
@GrahamCampbell have patience 😅

**Comment by driesvints** [comment]
I will change it once Till has tagged a new release.

File: composer.json (hunk 2 above)
Comment (tillkruss) [reply composer.json:92]: I'd agree with testing `v1.1.x-dev` first before I tag, just in case there is anther minor compat issue.

**Pull request merged**